test/e2e/util/util.go

```diff
@@ -1531,185 +1531,186 @@ func EnsureGuestWebhooksValidated(t *testing.T, ctx context.Context, guestClient
 }
 
 func EnsureGlobalPullSecret(t *testing.T, ctx context.Context, mgmtClient crclient.Client, entryHostedCluster *hyperv1.HostedCluster) error {
-	AtLeast(t, Version420)
-
-	if entryHostedCluster.Spec.Platform.Type != hyperv1.AzurePlatform {
-		t.Skip("test only supported on platform ARO")
-	}
-
-	var (
-		dummyImageTagMultiarch = "quay.io/hypershift/sleep:multiarch"
-		dummyImageTag12        = "quay.io/hypershift/sleep:1.2.0"
-		err                    error
-
-		// Additional Pull Secret
-		additionalPullSecretName            = "additional-pull-secret"
-		additionalPullSecretNamespace       = "kube-system"
-		pullSecretNamespace                 = "openshift-config"
-		additionalPullSecretDummyData       = []byte(`{"auths": {"quay.io": {"auth": "YWRtaW46cGFzc3dvcmQ="}}}`)
-		additionalPullSecretReadOnlyE2EData = []byte(`{"auths": {"quay.io": {"auth": "aHlwZXJzaGlmdCtlMmVfcmVhZG9ubHk6R1U2V0ZDTzVaVkJHVDJPREE1VVAxT0lCOVlNMFg2TlY0UkZCT1lJSjE3TDBWOFpTVlFGVE5BS0daNTNNQVAzRA=="}}}`)
-		oldglobalPullSecretData             []byte
-		dsImage                             string
-	)
-	g := NewWithT(t)
+	t.Run("EnsureGlobalPullSecret", func(t *testing.T) {
+		AtLeast(t, Version419)
+		if entryHostedCluster.Spec.Platform.Type != hyperv1.AzurePlatform {
+			t.Skip("test only supported on platform ARO")
```

Comment:
@jparrill this will break down the day we'll support not managed on Azure. To fix this I would consider Is https://github.com/openshift/hypershift/blob/1b91a8234cc4383f59e14dfd1ae22ae52a0cb8df/support/azureutil/azureutil.go#L186

Reply:
Yes and no, Bryan is also working in self-managed azure, this is why I've followed the platform.Type check instead of ARO.